### ISSUE TITLE
Classloader mismatches

### ISSUE THREAD

**Issue by Rhobal** [GoogleCloudPlatform/google-cloud-eclipse#1971]
TLDR: The App Engine Standard Runtime brings a lot of Jars that are not visible to Eclipse and are preferred over Jars on the build path
----
Trying to debug a class loader issue with `javax.el.ExpressionFactory#newInstance()`, I opened a new HelloAppEngine Maven project and I quickly ran into classloader issues:

With the default pom.xml, Eclipse takes the ExpressionFactory from  jsp-api-2.1.jar which doesn't contain `newInstance()`

If I add a dependency to `javax.el:javax.el-api:jar:3.0.0` to the pom, the ExpressionFactory in there does contain `newInstance()` but on execution, the ExpressionFactory is taken from Cloud SDK/google-cloud-sdk/platform/google_appengine/google/appengine/tools/java/lib/shared/el-api.jar
and newInstance fails with a `ClassNotFoundException: de.odysseus.el.ExpressionFactoryImpl`

In the Cloud SDK path, there are a lot of Jars that are not visible in eclipse and there is no obvious way to override them. Plus the el-api.jar seems incomplete as it references a class that is not available on the classpath.

**Comment by chanseokoh** [comment]
> The App Engine Standard Runtime brings a lot of Jars that are not visible to Eclipse

The `jsp-api-2.1.jar` is actually visible, in the _Package Explorer_ or on the _Java Build Path_ project preference page. (Expand _App Engine Standard Runtime_.)

![selection_012](https://cloud.githubusercontent.com/assets/10523105/26253593/eb028432-3c81-11e7-9e3a-f1d5ae57781f.png)

That said, [the Google App Engine standard environment is a "sandboxed" environment, and when using Java 7, it uses Servlet 2.5](https://cloud.google.com/appengine/docs/standard/java/runtime). I believe EL 2.1 is a component of the JSP 2.1 Specification (http://www.bigdev.org/2012/01/jsf-version-dependencies-on-servlet-jsp-el-container/), and I _think_ you want to be at or above Servlet 3.1 to use EL 3.0.

If you still want to use EL 3.0 (hence Servlet 3.1), you can try the GAE flexible environment, or wait a bit until Java8 + Servlet 3.1 becomes generally available on GAE.
